RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc

```diff
@@ -94,20 +111,18 @@ void SiPixelDigisClustersFromSoA::produce(edm::StreamID, edm::Event& iEvent, con
         edm::LogWarning("SiPixelDigisClustersFromSoA") << "cluster below charge Threshold "
                                                        << "Layer/DetId/clusId " << layer << '/' << detId << '/' << ic
                                                        << " size/charge " << acluster.isize << '/' << acluster.charge;
-      SiPixelCluster cluster(acluster.isize, acluster.adc, acluster.x, acluster.y, acluster.xmin, acluster.ymin, ic);
-#ifdef EDM_ML_DEBUG
-      ++totClustersFilled;
-#endif
-      LogDebug("SiPixelDigisClustersFromSoA")
-          << "putting in this cluster " << ic << " " << cluster.charge() << " " << cluster.pixelADC().size();
       // sort by row (x)
-      spc.push_back(std::move(cluster));
+      spc.emplace_back(acluster.isize, acluster.adc, acluster.x, acluster.y, acluster.xmin, acluster.ymin, ic);
+      aclusters[ic].clear();
       std::push_heap(spc.begin(), spc.end(), [](SiPixelCluster const& cl1, SiPixelCluster const& cl2) {
         return cl1.minPixelRow() < cl2.minPixelRow();
       });
+#ifdef EDM_ML_DEBUG
+      ++totClustersFilled;
+      LogDebug("SiPixelDigisClustersFromSoA")
+          << "putting in this cluster " << ic << " " << cluster.charge() << " " << cluster.pixelADC().size();
```

Comment:
@VinInn , this failed to build for DBG IBs (built with `-DEDM_ML_DEBUG`). Can you please provide a fix? 

[a] https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc900/CMSSW_12_0_DBG_X_2021-07-01-2300/RecoLocalTracker/SiPixelClusterizer
```
  RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:123:57: error: 'cluster' was not declared in this scope; did you mean 'acluster'?
   123 |           << "putting in this cluster " << ic << " " << cluster.charge() << " " << cluster.pixelADC().size();
      |                                                         ^~~~~~~
      |                                                         acluster
```

Reply:
@smuzaffar @VinInn please have a look at https://github.com/cms-sw/cmssw/pull/34310 (if that's the desired behavior). Based on the struct 
https://github.com/cms-sw/cmssw/blob/77ba30c7d3f6c3b3dc0db63e7cdf847904c627e9/RecoLocalTracker/SiPixelClusterizer/plugins/PixelClusterizerBase.h#L18-L58

looks like it is.